static.config.js

```diff
@@ -41,249 +30,90 @@ import {
-
-  // get parent, either if its topics as parent or department
-  if (
-    parent_topic &&
-    grandparent_topic_collection &&
-    allTopics &&
-    allTopics.edges.length &&
-    allTopicCollections &&
-    allTopicCollections.edges.length &&
-    allTopicCollections.edges[0].node.theme
-  ) {
-    contextualNavData.parent = {
-      id: allTopics.edges[0].node.id,
-      title: allTopics.edges[0].node.title,
-      // url is theme/topic-collection/topic
-      url: `/${allTopicCollections.edges[0].node.theme.slug}/${
-        allTopicCollections.edges[0].node.slug
-      }/${allTopics.edges[0].node.slug}/`,
-    };
-  }
-
-  if (
-    parent_department &&
-    allDepartmentPages && // note, not all dept pages, the result of the query
-    allDepartmentPages.edges.length
-  ) {
-    contextualNavData.parent = {
-      id: allDepartmentPages.edges[0].node.id,
-      title: allDepartmentPages.edges[0].node.title,
-      url: `/${allDepartmentPages.edges[0].node.slug}/`,
-    };
-  }
+  const { topicCollectionTopics } = await client.request(
+    getTopicCollectionTopicQuery,
+    {
+      id: grandparent.id,
+    },
+  );
 
-  // get related to
   if (
-    parent_topic &&
-    grandparent_topic_collection &&
     topicCollectionTopics &&
     topicCollectionTopics.edges.length &&
     allTopicCollections &&
-    allTopicCollections.edges.length &&
-    allTopicCollections.edges[0].node.theme
+    allTopicCollections.edges.length
   ) {
-    contextualNavData.relatedTo = topicCollectionTopics.edges
-      .filter(edge => edge.node && edge.node.page.topicpage.id !== parent_topic)
+    relatedTo = topicCollectionTopics.edges
+      .filter(edge => edge.node && edge.node.page.topicpage.id !== parent.id)
       .map(edge => ({
         id: edge.node.page.topicpage.id,
         title: edge.node.page.topicpage.title,
-        url: `/${allTopicCollections.edges[0].node.theme.slug}/${
-          allTopicCollections.edges[0].node.slug
-        }/${edge.node.page.topicpage.slug}/`,
+        url: `/${grandparent.url}/${edge.node.page.topicpage.slug}/`,
       }));
-  } else {
-    contextualNavData.relatedTo = [];
-    // relatedTo is empty if we are viewing page under the department
-  }
-
-  // get offered by
-  console.log('departments from cn: ', departments);
-  contextualNavData.offeredBy = getOfferedByFromDepartments(departments);
-
-  return contextualNavData;
-};
-
-const getAllTopicLinks = (
-  allServicePageTopics,
-  allInformationPageTopics,
-  allOfficialDocumentPageTopics,
-  allGuidePageTopics,
-  allFormContainerTopics,
-) => {
-  /* 
-    getAllTopicLinks: collects all the topic links from servicepagetopics, infopage topics
-    official doc topics, guidepage topics and form container topics into one 
-    array of links.
-    Used in getTopicPageData for topics.otherLinks
-    --> gets the pages that aren't top pages. so the other pages. 
-  */
-  // I don't like this but we still need to do some logic here
-  // to get all the pages
```

Comment:
so glad we don't need to do this anymore!

Reply:
same! thanks for writing that resolver